### ISSUE TITLE
Add support to type annotations

### ISSUE THREAD

**Issue by pgrandjean** [propensive/magnolia#254]
According to the scala [documentation](https://www.scala-lang.org/files/archive/spec/2.12/11-annotations.html), annotations may apply to classes, variables, types and expressions. Currently, Magnolia allows to extract variable annotations from case classes, such as:
```scala
case class Foo(
  @Bar field1: String
)
```

The aim of this request is to add support for type annotations, such as:
```scala
case class Foo(
  field1: String @Bar
)
```

**Comment by propensive** [comment]
It's a good suggestion, @pgrandjean, and I don't think it would be too difficult to add it to Magnolia's interface as a `typeAnnotations` field on all of `CaseClass`, `SealedTrait`, `Parameter` and `Subtype`. Would anyone like to work on it?

A PR introducing a not-yet-compiling test case would be very welcome in the meantime.

**Comment by pgrandjean** [comment]
@propensive I started working on it (cf PR#257). Started with `Parameter`.

**Comment by joroKr21** [comment]
Fixed by #257
